### ISSUE TITLE
parents not set for all objects

### ISSUE THREAD

**Issue by rob-c** [ganga-devs/ganga#438]
From my LSST work I see the following error:

```
WARNING  ---------- error in user/extension code ----------
WARNING  Traceback (most recent call last):
WARNING    File "/afs/cern.ch/user/r/rcurrie/cmtuser/GANGA/GANGA_LSST/install/ganga/python/Ganga/GPIDev/Lib/Job/Job.py", line 621, in updateStatus
WARNING      getattr(self, state.hook)()
WARNING    File "/afs/cern.ch/user/r/rcurrie/cmtuser/GANGA/GANGA_LSST/install/ganga/python/Ganga/GPIDev/Lib/Job/Job.py", line 851, in postprocess_hook
WARNING      self.postprocessoutput(self.outputfiles, self.outputdir)
WARNING    File "/afs/cern.ch/user/r/rcurrie/cmtuser/GANGA/GANGA_LSST/install/ganga/python/Ganga/GPIDev/Lib/Job/Job.py", line 739, in postprocessoutput
WARNING      outputfile.processOutputWildcardMatches()
WARNING    File "/afs/cern.ch/user/r/rcurrie/cmtuser/GANGA/GANGA_LSST/install/ganga/python/Ganga/GPIDev/Lib/File/LocalFile.py", line 117, in processOutputWildcardMatches
WARNING      sourceDir = self.getJobObject().outputdir
WARNING    File "/afs/cern.ch/user/r/rcurrie/cmtuser/GANGA/GANGA_LSST/install/ganga/python/Ganga/GPIDev/Base/Objects.py", line 1077, in getJobObject
WARNING      raise AssertionError('no job associated with object ' + repr(self))
WARNING  AssertionError: no job associated with object LocalFile(namePattern='*.txt')
WARNING  --------------------------------------------------
WARNING  An error occured finalising job: 179
WARNING  Attemting again (5 of 5) after 2.5-sec delay
```

I see this after copying a ```completed``` job with a ```LocalFile(namePattern="*.txt")``` in the outputfiles (which hasn't resolved either)

I suspect I'm seeing (possibly) 2 bugs:
 1. Multiple ```*.ext``` filetypes in the same outputfiles aren't populated
 2. Some files in the ```outputfiles``` are somehow not having their parent set correctly.

I'm only running with a new plugin + #430 and as far as I can see this PR shouldn't cause the problems I'm seeing here.

**Comment by rob-c** [comment]
```
j=Job(application=Executable(exe='generateTest.sh'), backend=Dirac(), inputfiles=[LocalFile('~/cmtuser/GANGA/GANGA_LSST/install/ganga/generateTest.py'), LocalFile('/afs/cern.ch/user/r/rcurrie/cmtuser/GANGA/GANGA_LSST/install/ganga/generateTest.sh')]); j.outputfiles=[LocalFile('*.txt'), DiracFile('*.root'), LocalFile('*.zip')]; j.submit()
```
and
```
j=Job(application=Executable(exe='generateTest.sh'), backend=Dirac(), inputfiles=[LocalFile('~/cmtuser/GANGA/GANGA_LSST/install/ganga/generateTest.py'), LocalFile('/afs/cern.ch/user/r/rcurrie/cmtuser/GANGA/GANGA_LSST/install/ganga/generateTest.sh')]); j.outputfiles=[LocalFile('*.txt')]; j.submit()
```
both work as expected when launched/completed in the same ganga session.

I'll try re-running after loading from disk and after making a copy to see what may be going wrong.

**Comment by rob-c** [comment]
Hmmm, can't faithfully reproduce this... 

If this crops up again I'll try to dig at what happened but after some 10 or so jobs in different configurations generated in different ways this doesn't come back. I'm putting this down to an oddity.

I'd like to leave this open for a week or so but if I can't reproduce it then I'll close this

**Comment by milliams** [comment]
@rob-c Have you seen this again at all? If not we should perhaps close it.

**Comment by rob-c** [comment]
Haven't in a while but I think I know what caused it and it's not develop code.

**Comment by rob-c** [comment]
Rats, Just observed:

```

**Comment by rob-c** [comment]
Job was submitted as:
```
j=Job(application=Executable(exe=File('~/cmtuser/GANGA/GANGA_LSST/install/ganga/generateTest.py')), backend=Dirac(), inputfiles=[LocalFile('~/cmtuser/GANGA/GANGA_LSST/install/ganga/generateTest.py'), LocalFile('/afs/cern.ch/user/r/rcurrie/cmtuser/GANGA/GANGA_LSST/install/ganga/generateTest.sh')]); j.outputfiles=[LocalFile('*.txt'), DiracFile('*.root'), LocalFile('*.zip')]; j.submit()
```
I waited 5min as I was testing something else and then this error occurs...

Aside: I don't think we should stop parsing when just 1 of the patterns isn't matched. This could leave untracked large files on the grid or elsewhere if the user does something daft and mis-types a ```LocalFile pattern```. We should attempt to match all file patterns imo then mark the job as failed anyway.

**Comment by rob-c** [comment]
This could be caused by the bugs I fixed in #473 which would cause confusion like this as the job could be entering the finalize step multiple times quite quickly if the monitoring is doing it's job.

**Comment by rob-c** [comment]
This doesn't appear to be a problem when I close ganga and perform a ```job.backend.reset()``` as Ganga completes the job as expected.
I'm now testing #473 against this problem.
